### PR TITLE
[CALCITE-4367] Fix RpcMetadata, DatabasePropertyRequest, et al. documentation (john-bodley)

### PR DESCRIPTION
**Pull request by john-bodley**
See [CALCITE-4367](https://issues.apache.org/jira/browse/CALCITE-4367) for specifics.

### CODE REVIEW

**Review by danny0405 — APPROVED** [review]
+1, thanks for the fix.

**Comment by danny0405** [comment]
Can you log an issue there so that it can be tracked by the JIRA.

**Comment by john-bodley** [comment]
@danny0405 I also noticed a few values which were not marked as optional but the Avatica JDBC driver presented as `null`.

**Comment by danny0405** [comment]
> @danny0405 I also noticed a few values which were not marked as optional but the Avatica JDBC driver presented as `null`.

Thanks, please fix it also and update the commit message as "[CALCITE-{JIRA number}] main message (your github account name)".

File: site/_docs/json_reference.md [hunk 1]
```diff
@@ -920,6 +934,7 @@ This object represents the properties for a given JDBC Connection.
 {% highlight json %}
 {
   "connProps": "connPropsImpl",
+  "dirty": true,
```
Comment (john-bodley) [inline site/_docs/json_reference.md:937]: This is [`is_dirty`](https://github.com/apache/calcite-avatica/blob/4b7eee5bf430b916c7c07897b6f60d2b6b6dabb7/core/src/main/protobuf/common.proto#L24) for protobuf but seems to be merely `dirty` for JSON.

**Comment by john-bodley** [comment]
@danny0405 I've made the updates you requested.

**Comment by danny0405** [comment]
> @danny0405 I've made the updates you requested.

Hmm, you missed the GitHub account, and please also squish the commits into one.

**Comment by F21** [comment]
@danny0405 Can you please take a look at this?

**Comment by joshelser** [comment]
@F21 @danny0405 what's going on here? We have two pull-requests for this one Jira issue? see #131

**Comment by F21** [comment]
hmm. This one has a few more changes in addition to the changes from #131.

**Comment by joshelser** [comment]
> This one has a few more changes in addition to the changes..

Can we get a new Jira issue open before any changes get merged, please? Will be good to keep the release note accurate 🙏🏼

**Comment by F21** [comment]
@joshelser After looking at your second comment in the jira issue, perhaps the changes in this PR are not needed? Can you please confirm?

**Comment by joshelser** [comment]
>  Can you please confirm?

Glancing through a couple of the changes, I think I re-implemented all of them. The author didn't link this PR to the Jira issue, so I had no idea he actually made these changes :(. I think this can just be closed.

**Comment by F21** [comment]
Thanks! I'll go ahead and close this.

**Pull request closed without merging**